addons/purchase/models/purchase_order.py

```diff
@@ -708,17 +701,32 @@ def action_create_invoice(self):
         invoice_vals_list = new_invoice_vals_list
 
         # 3) Create invoices.
-        moves = self.env['account.move']
+        invoices = self.env['account.move']
         AccountMove = self.env['account.move'].with_context(default_move_type='in_invoice')
         for vals in invoice_vals_list:
-            moves |= AccountMove.with_company(vals['company_id']).create(vals)
+            invoices |= AccountMove.with_company(vals['company_id']).create(vals)
 
         # 4) Some moves might actually be refunds: convert them if the total amount is negative
         # We do this after the moves have been created since we need taxes, etc. to know if the total
         # is actually negative or not
-        moves.filtered(lambda m: m.currency_id.round(m.amount_total) < 0).action_switch_move_type()
+        invoices.filtered(lambda m: m.currency_id.round(m.amount_total) < 0).action_switch_move_type()
+
+        # 5) Link the attachments to the invoice
+        attachments = self.env['ir.attachment'].browse(attachment_ids)
+        if not attachments:
+            return self.action_view_invoice(invoices)
+
+        if len(invoices) != 1:
+            raise ValidationError(_("You can only upload a bill for a single partner at a time."))
```

Comment:
The message is a bit misleading to me. should be 
"You can only upload a bill for a single purchase order"